zowe/zowe-cli
# Daemon Mode PoC

Planned updates to allow running Zowe CLI as a persistent process (~daemon) for faster response time and a new native client is introduced to communicate with the persistent process.  Implementation, behavior, command names are all subject to change.  

To test:
- Obtain `zowex.exe` binary for your platform and add to `PATH`
- Run `zowe` as a background task with a hidden `--daemon` option:
  - Windows `start zowe --daemon`
  - Linux `zowe --daemon &`
  - pm2 `pm2 start <global-npm-location>/node_modules/@zowe/cli/lib/main.js --name zowe-daemon -- --daemon`
    - `npm list -g | head -1` to see `<global-npm-location>`

Daemon port is configurable via `ZOWE_DAEMON` - the default port is `4000`

Then run commands using `zowex` with the same parameters that you normally use e.g. `zowex jobs list jobs --help`.  

**Known Issues**
* `prompt*` does not work
* `login` does not work
